tests/dash/proto_utils.py

```diff
@@ -94,6 +99,70 @@ def routing_type_from_json(json_obj):
     return pb
 
 
+def acl_group_from_json(json_obj):
+    pb = AclGroup()
+    pb.guid.value = bytes.fromhex(uuid.UUID(json_obj["guid"]).hex)
+    pb.ip_version = IpVersion.IP_VERSION_IPV4
+    return pb
+
+
+def acl_out_from_json(json_obj):
+    pb = AclOut()
+    pb.v4_acl_group_id = json_obj["acl_group_id"]
+    return pb
+
+
+def acl_in_from_json(json_obj):
+    pb = AclIn()
+    pb.v4_acl_group_id = json_obj["acl_group_id"]
+    return pb
+
+
+def acl_rule_from_json(json_obj):
+    pb = AclRule()
+    pb.priority = int(json_obj["priority"])
+    pb.action = Action.ACTION_DENY if json_obj["action"] == "deny" else Action.ACTION_PERMIT
+    pb.terminating = json_obj["terminating"] == "true"
+    if "src_addr" in json_obj:
+        for addr in json_obj["src_addr"].split(','):
+            net = ipaddress.IPv4Network(addr, False)
+            ip = IpPrefix()
+            ip.ip.ipv4 = socket.htonl(int(net.network_address))
+            ip.mask.ipv4 = socket.htonl(int(net.netmask))
+            pb.src_addr.append(ip)
+    if "dst_addr" in json_obj:
+        for addr in json_obj["dst_addr"].split(','):
+            net = ipaddress.IPv4Network(addr, False)
+            ip = IpPrefix()
+            ip.ip.ipv4 = socket.htonl(int(net.network_address))
+            ip.mask.ipv4 = socket.htonl(int(net.netmask))
+            pb.dst_addr.append(ip)
+    elif "src_port" in json_obj:
```

Comment:
Shouldn't it be `if` instead of `elif` for this field and the fields below?
With `elif` the fields will get ignored if the "dst_addr" is specified.

Reply:
Yes, you are right. Thanks for your suggestion. Fixed.